lib/src/main/java/org/esbtools/lightbluenotificationhook/NotificationHookConfiguration.java

```diff
@@ -6,55 +6,100 @@
 
 import java.util.Objects;
 
+/**
+ * Configures the notification hook for an entity
+ *
+ * watchProjection: This projection defines a part of the document
+ * that is watched for changes. Any modifications to the fields
+ * included in this projection will result in a notification being
+ * fired
+ *
+ * includeProjection: This projection defines the payload of the
+ * message.
+ *
+ * propertiesProjection: This optional projection selects some fields
+ * in the entity to be stored as name-value pairs in properties field.
+ *
+ * arrayOrderingSignificant: If this is set to true, and if the only
+ * thing changed in an array is the ordering of its elements, an event
+ * is still generated. If false, array re-orderings are
+ * ignored. Default is false.
+ */
 public class NotificationHookConfiguration implements HookConfiguration {
     private final Projection watchProjection;
     private final Projection includeProjection;
+    private final Projection propertiesProjection;
+    private final boolean arrayOrderingSignificant;
 
-    public NotificationHookConfiguration(Projection watchProjection, Projection includeProjection) {
+    public NotificationHookConfiguration(Projection watchProjection,
+                                         Projection includeProjection,
+                                         Projection propertiesProjection,
+                                         boolean arrayOrderingSignificant) {
```

Comment:
I prefer enums to bool flags, but it's not a huge deal since this is such a small project.